code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -1,45 +1,48 @@
 //Engineering Mesons
 
+#define MODE_NONE ""
+#define MODE_MESON "meson"
+#define MODE_TRAY "t-ray"
+#define MODE_RAD "radiation"
+
 /obj/item/clothing/glasses/meson/engine
 	name = "engineering scanner goggles"
 	desc = "Goggles used by engineers. The Meson Scanner mode lets you see basic structural and terrain layouts through walls, regardless of lighting condition. The T-ray Scanner mode lets you see underfloor objects such as cables and pipes."
-	icon_state = "trayson-meson"
+	icon_state = "trayson"
 	actions_types = list(/datum/action/item_action/toggle_mode)
 	origin_tech = "materials=3;magnets=3;engineering=3;plasmatech=3"
 
-	var/mesons_on = TRUE //if set to FALSE, these goggles work as t-ray scanners.
-	var/range = 1
-
+	vision_flags = 0
+	darkness_view = 2
+	invis_view = SEE_INVISIBLE_LIVING
 
+	var/list/modes = list(MODE_NONE = MODE_MESON, MODE_MESON = MODE_TRAY, MODE_TRAY = MODE_RAD, MODE_RAD = MODE_NONE)
+	var/mode = MODE_NONE //if set to FALSE, these goggles work as t-ray scanners.
```

Comment:
remove the comment if it's no longer relevant

Reply:
oh woops